### PR TITLE
Get Largest Output for tree and for forest

### PR DESCRIPTION
**Pull request by sslavian812**
Related to issue automl/random_forest_run#45.
 - Added method providing largest possible output of the tree.
 - Added method providing list of largest possible values from all trees in the forest

### CODE REVIEW

File: include/rfr/trees/k_ary_tree.hpp [hunk 1]
```diff
@@ -172,6 +172,21 @@ class k_ary_random_tree : public rfr::trees::tree_base<num_t, response_t, index_
 		return(the_nodes[node_index]);
 	}
 
+	num_t get_largest_output() const {
+		num_t cur_max = -1000*1000*1000; // -10^9 should be enough
```
Comment (sfalkner) [inline include/rfr/trees/k_ary_tree.hpp]: A true 'infinity' value should be used here!

File: include/rfr/trees/k_ary_tree.hpp [hunk 2]
```diff
@@ -172,6 +172,21 @@ class k_ary_random_tree : public rfr::trees::tree_base<num_t, response_t, index_
...
+		bool first = true;
+
+		for(int i=0; i<the_nodes.size(); ++i) {
+			if(the_nodes[i].is_a_leaf()){
+				rfr::util::weighted_running_statistics<num_t> const& st = the_nodes[i].leaf_statistic();
+				if(cur_max <= st.mean() || first){
```
Comment (sfalkner) [inline include/rfr/trees/k_ary_tree.hpp]: You don't have to check if that is the first pass, if you use a true 'inf' value above. This reads cumbersome.

**Comment by sfalkner** [comment]
Thanks for the pull request. I added a few comments in your code. The major problem I have, is that there is no unit test for you code. Without those, I won't merge the pull request, sorry.

**Comment by eddiebergman** [comment]
Going to close as this has gone stale

**Pull request closed without merging**